pypeit/core/procimg.py

```diff
@@ -912,96 +913,271 @@ def trim_frame(frame, mask):
+          read-noise squared),
+        - :math:`V_{\rm proc}` is added variance from image processing (e.g.,
+          bias subtraction), and
+        - :math:`\epsilon` is an added error term that imposes a maximum
+          signal-to-noise on the observed counts (see ``noise_floor``).
 
+    The function :func:`base_variance` consolidates all terms
+    that do not change with the forward
+    modeling of the sky + object counts into a single "base-level" variance
 
+    .. math::
+
+        V_{\rm base} = s^2\ \left[ D t_{\rm exp} / 3600 + V_{\rm rn} + V_{\rm
+        proc} \right]
+
+    such that the first equation can be re-written as
+
+    .. math::
+
+        V = s {\rm max}(0,c) + V_{\rm base} + \epsilon^2 {\rm max}(0, c)^2,
+
+    which is the quantity returned by this function.
+
+    We emphasize that this is a *model* for the per-pixel image variance.  In
+    real data, the as-observed pixel values are used to estimate the Poisson
+    error in the observed counts.  Because of the variance in the image, this
+    systematically overestimates the variance toward low counts (:math:`\lesssim
+    2 \sigma_{\rm rn}`), with a bias of approximately :math:`1.4/\sigma_{\rm
+    rn}` for :math:`C=0` (i.e., about 20% for a readnoise of 2 e-) and less than
+    10% for :math:`C=1`.
+
+    .. note::
+
+        If :math:`s` (``count_scale``) is provided, the variance will be 0
+        wherever :math:`s \leq 0`, modulo the provided ``noise_floor``.
+
+    Args:
+        base (`numpy.ndarray`_):
+            The "base-level" variance in the data set by the detector properties
+            and the image processing steps.  See :func:`base_variance`;
+            :math:`V_{\rm base}` in the equations above.
+        counts (`numpy.ndarray`_, optional):
+            A 2D array with the number of source-plus-sky counts, possibly
+            rescaled by a relative throughput; see :math:`c` in the equations
+            above.  Because this is used to calculate (part of) the Poisson
+            statistics for the electon counts and the noise floor, this *must*
+            be provided if ``noise_floor`` is not None or ``shot_noise`` is
+            True.  Shape must match ``base``.
+        count_scale (:obj:`float`, `numpy.ndarray`_, optional):
+            A scale factor that *has already been applied* to the provided
+            counts; see :math:`s` in the equations above.  For example, if the
+            image has been flat-field corrected, this is the inverse of the
+            flat-field counts.  If None, no scaling is expected, meaning
+            ``counts`` are exactly the observed detector counts.  If a single
+            float, assumed to be constant across the full image.  If an array,
+            the shape must match ``base``.  The variance will be 0 wherever
+            :math:`s \leq 0`, modulo the provided ``noise_floor``.
+        noise_floor (:obj:`float`, optional):
```

Comment:
Should we define this in terms of SNR_clip = 1/noise_floor instead? To me that is more clear, i.e. it is the maximum S/N we enforce.

Reply:
Done.